google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/PartitionCountWatchingPublisher.java

```diff
@@ -94,11 +98,13 @@ public void stop() {
   @GuardedBy("monitor.monitor")
   private Optional<PartitionsWithRouting> partitionsWithRouting = Optional.empty();
 
-  public PartitionCountWatchingPublisher(PartitionCountWatchingPublisherSettings settings) {
-    this.publisherFactory = settings.publisherFactory();
-    this.policyFactory = settings.routingPolicyFactory();
-    PartitionCountWatcher configWatcher =
-        settings.configWatcherFactory().newWatcher(this::handleConfig);
+  PartitionCountWatchingPublisher(
+      PartitionPublisherFactory publisherFactory,
+      RoutingPolicy.Factory policyFactory,
+      PartitionCountWatcher.Factory configWatcherFactory) {
+    this.publisherFactory = publisherFactory;
+    this.policyFactory = policyFactory;
+    PartitionCountWatcher configWatcher = configWatcherFactory.newWatcher(this::handleConfig);
     addServices(configWatcher);
   }
```

Comment:
Memo for myself. This constructor is now invisible but used by Beam
https://github.com/apache/beam/blob/243128a8fc52798e1b58b0cf1a271d95ee7aa241/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/Publishers.java#L46